From be8e29ec073e09bf94efa382f720b0b78d3f85f0 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Sun, 28 Apr 2019 11:19:47 +0300
Subject: [PATCH] [SW-0] preboot: consolidate declarations

There are several declarations that are used by preboot common code,
but for some reason declared (and several even implemented) in
goya/gaudi specific headers.
This restricts the headers inclusion flexibility and make the
maintenance burden bigger.

Move the declarations into the common header and move the
implementations into the .c files.

Change-Id: I845387f423effb1b7d1ca632e4668b7edce631e0
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/common/preboot.h b/common/preboot.h
index ecc353e..7995125 100644
--- a/common/preboot.h
+++ b/common/preboot.h
@@ -48,4 +48,8 @@
 void hl_uart_init(void);
 u32 hl_spi_baudr_get(void);
 
+void preboot_pcie_wait_for_link(void);
+void preboot_delay(void);
+void preboot_putc_virt_uart(const char c);
+
 #endif /* COMMON_PREBOOT_H_ */
diff --git a/gaudi/gaudi.c b/gaudi/gaudi.c
index 6068038..87a55bb 100644
--- a/gaudi/gaudi.c
+++ b/gaudi/gaudi.c
@@ -24,6 +24,9 @@
 	}
 }
 
+void preboot_delay(void) {}
+void preboot_putc_virt_uart(const char c) {}
+
 u64 get_cs_pad_sel(void)
 {
 	return mmPSOC_GLOBAL_CONF_PAD_SEL_53;
diff --git a/gaudi/gaudi.h b/gaudi/gaudi.h
index 443573e..7e2e70a 100644
--- a/gaudi/gaudi.h
+++ b/gaudi/gaudi.h
@@ -37,8 +37,3 @@
 #define SPI_BAUDR			(PCI_DIV2_CLOCK / 25000000)
 
 #define NON_RST_FLOPS_3			mmPSOC_GLOBAL_CONF_COLD_RST_FLOPS_3
-
-void preboot_pcie_wait_for_link(void);
-static inline void preboot_delay(void) {}
-static inline void preboot_putc_virt_uart(const char c) {}
-
diff --git a/goya/goya.h b/goya/goya.h
index 1478364..ac4ff5a 100644
--- a/goya/goya.h
+++ b/goya/goya.h
@@ -15,7 +15,3 @@
 #define HL_CPU_CYCLES_PER_USEC	133000000
 
 #define NON_RST_FLOPS_3		mmPSOC_GLOBAL_CONF_NON_RST_FLOPS_3
-
-void preboot_pcie_wait_for_link(void);
-void preboot_delay(void);
-void preboot_putc_virt_uart(const char c);
